openquake/engine/engine.py

```diff
@@ -109,23 +109,34 @@ def job_stats(job):
     js = job.jobstats
     try:
         yield
-    except:
-        conn = django_db.connections['job_init']
-        if conn.is_dirty():
-            conn.rollback()
-        raise
     finally:
+        tb = traceback.format_exc()  # get the traceback of the error, if any
         job.is_running = False
-        job.save()
-
-        # save job stats
-        curs.execute("select pg_database_size(%s)", (dbname,))
-        new_dbsize = curs.fetchall()[0][0]
-        js.disk_space = new_dbsize - dbsize
-        js.stop_time = datetime.utcnow()
-        js.save()
+        if tb != 'None\n':
+            # rollback the transactions; unfortunately, for mysterious reasons,
+            # this is not enough and an OperationError may still show up in the
+            # finalization phase when forks are involved
+            for conn in django_db.connections.all():
+                conn.rollback()
```

Comment:
Before `conn` was taken from `django_db.connections['job_init']`; now this has been removed. Where `conn` is from?